### ISSUE TITLE
Dropout variables are inconsistent across save/load

### ISSUE THREAD

**Issue by LarsKue** [keras-team/keras#19796]
`keras.layers.Dropout` holds just a single variable, the `seed_generator_state`. This variable changes when the model is saved and loaded from disk. Notably, this also affects the behavior of the layer:

```py3
import os
os.environ["KERAS_BACKEND"] = "torch"

import keras


def test_serialize_deserialize(tmp_path):
    model = keras.Sequential([keras.layers.Dropout(0.5)])

    model(keras.Input(shape=(128,)))

    keras.saving.save_model(model, tmp_path / "model.keras")
    loaded = keras.saving.load_model(tmp_path / "model.keras")

    x = keras.random.normal(shape=(128, 128))

    x1 = model(x, training=True)
    x2 = loaded(x, training=True)

    # fails
    assert keras.ops.all(keras.ops.isclose(x1, x2))

    for v1, v2 in zip(model.variables, loaded.variables):
        v1 = keras.ops.convert_to_numpy(v1)
        v2 = keras.ops.convert_to_numpy(v2)
        
        # here for demonstration
        print(v1, v2)
        
        # also fails
        assert keras.ops.all(keras.ops.isclose(v1, v2))
```

Note I only tested it on the torch backend. It looks like the perpetrator is that this variable is not saved in the config:

```py3
# file: keras/src/layers/regularization/dropout.py
class Dropout(Layer):
    def get_config(self):
        base_config = super().get_config()
        config = {
            "rate": self.rate,
            "seed": self.seed,
            "noise_shape": self.noise_shape,
        }
        return {**base_config, **config}

```

Note that `seed` is a different variable, not saved in `self.variables`.

If this behavior is intended, then I am wondering how we should compare saved/loaded layers to verify they save properly. My use case is an automated test that compares all variables (among other things) to verify this.

**Comment by fchollet** [comment]
The Keras saving format isn't currently set up to save and load random seeds. That's not part of the API contract. It's something that could be added, though.

For now you can simply make your Dropout layer deterministic by passing `seed=...` (an int) to the constructor.

**Comment by fchollet** [comment]
> The Keras saving format isn't currently set up to save and load random seeds. 

To provide a bit more info here -- saving and loading work by walking the model structure and entering objects that are considered "trackable". `SeedGenerator` objects aren't considered trackable at this time. It would be easy to make them trackable, but a potential downside is that adding/removing a `SeedGenerator` to a model would make loading fail.

**Comment by LarsKue** [comment]
I see, that sounds like there is currently no good fix. Instead, is there a better way to verify that the loaded model's variables are equivalent to the saved model's, without accidentally looping over untrackable variables or ones not part of the API contract?

**Comment by fchollet** [comment]
In this case, to check forward pass correctness, you can simply call the model in inference mode (`training=False`).

**Comment by mehtamansi29** [comment]
Hi @LarsKue -

Does this issue is currently reproducing ?
